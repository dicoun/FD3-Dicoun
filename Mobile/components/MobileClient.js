import React from 'react';
import PropTypes from 'prop-types';

import './MobileClient.css';

import {voteEvents} from './events';

class MobileClient extends React.PureComponent {

  static propTypes = {
    client: PropTypes.object.isRequired
  };

  state = {
    client: this.props.client
  };

  edit = (EO) => {
    voteEvents.emit('EditBtnClicked',this.props.client.id);
  }

  delete = (EO) => {
    voteEvents.emit('DeleteBtnClicked',this.props.client.id);
  }

  save = (EO) => {
    voteEvents.emit('SaveBtnClicked',this.props.client.id);
  }

  cancel = (EO) => {
    voteEvents.emit('CancelBtnClicked',this.props.client.id);
  }

  setSurnameRef = (ref) => {
    voteEvents.emit('setSurnameRef', ref);
  }

  setBallanceRef = (ref) => {
    voteEvents.emit('setBallanceRef', ref);
  }

  setNameRef = (ref) => {
    voteEvents.emit('setNameRef', ref);
  }

  setMiddleNameRef = (ref) => {
    voteEvents.emit('setMiddleNameRef', ref);
  }

  render() {
    console.log("MobileClient id="+this.props.client.id+" render");
    return (
      <tr className={this.props.client.id}>
        <td>
          {(this.props.client.disabledFlag) &&
            <input type='text' className='text' defaultValue={this.props.client.fam} ref={this.setSurnameRef} disabled/>
          }
          {(!this.props.client.disabledFlag ) &&
            <input type='text' className='text' defaultValue={this.props.client.fam} ref={this.setSurnameRef}/>
          }
        </td>
        <td>
          {(!this.props.client.addFlag) &&
              <input type='text' className='text' defaultValue={this.props.client.im} ref={this.setNameRef} disabled/>
          }
          {(this.props.client.addFlag) &&
              <input type='text' className='text' defaultValue={this.props.client.im} ref={this.setNameRef}/>
          }
        </td>
        <td>
          {(!this.props.client.addFlag) &&
            <input type='text' className='text' defaultValue={this.props.client.otch} ref={this.setMiddleNameRef} disabled/>
          }
          {(this.props.client.addFlag) &&
            <input type='text' className='text' defaultValue={this.props.client.otch} ref={this.setMiddleNameRef}/>
          }
        </td>
        <td>
          {(this.props.client.disabledFlag) &&
              <input type='text' className='text' defaultValue={this.props.client.balance} ref={this.setBallanceRef} disabled/>
          }
          {(!this.props.client.disabledFlag) &&
              <input type='text' className='text' defaultValue={this.props.client.balance} ref={this.setBallanceRef}/>
          }
        </td>
        {(this.props.client.state == 'active') && 
          <td className='StateActive'>
             <span className='State'>{this.state.client.state}</span>
          </td>
        }
        {(this.props.client.state == 'blocked') && 
          <td className='StateBlocked'>
             <span className='State'>{this.state.client.state}</span>
          </td>
        }   
        <td>
         {(this.props.client.disabledFlag) &&
            <input type='button' className='btnInGrid' value='Редактировать' onClick={this.edit}></input>
         }
         {(!this.props.client.disabledFlag) &&
            <input type='button' className='btnInGrid' value='Сохранить' onClick={this.save}></input>
         }
        </td>
        <td>
         {(this.props.client.disabledFlag) &&
            <input type='button' className='btnInGrid' value='Удалить' onClick={this.delete}></input>
         }
         {(!this.props.client.disabledFlag) &&
            <input type='button' className='btnInGrid' value='Отменить' onClick={this.cancel}></input>
         }
        </td>
      </tr>
    );
  }
}

export default MobileClient;
