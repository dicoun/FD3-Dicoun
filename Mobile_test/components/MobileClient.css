div.MobileClient
{
  padding: 2px 0;
}
span.MobileClientBalance
{
  float: right;
  width: 30px;
  font-weight: bold;
  text-align: right;
}
span.MobileClientFIO
{
  margin-right: 30px;
}
.text {
  width: 100%;
}
.StateActive{
  background-color:turquoise
}
.StateBlocked{
  background-color:tomato
}
input[type=text] {
  width: 90%;
}
.btnInGrid, .btnInGridEdit, .btnInGridDel{
  width: 150px;
}