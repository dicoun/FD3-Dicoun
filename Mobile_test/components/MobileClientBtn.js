import React from 'react'; 
import PropTypes from 'prop-types'; 
 
const MobileClientBtn = props => { 
  //return <div className='MobileClientBtn'>{props.question}</div>; 
  return <input type='button' data-kind={props.dataKind} className={props.className} value={props.value} onClick={props.onClick}></input> 
} 
 
MobileClientBtn.propTypes = { 
  dataKind: PropTypes.string.isRequired, 
  className: PropTypes.string.isRequired, 
  value: PropTypes.string.isRequired, 
  onClick: PropTypes.func.isRequired, 
}; 
 
export default MobileClientBtn;