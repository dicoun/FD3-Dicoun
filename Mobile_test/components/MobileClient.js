import React from 'react';
import PropTypes from 'prop-types';

import './MobileClient.css';
import MobileClientBtn from './MobileClientBtn.js';

import {voteEvents} from './events';

class MobileClient extends React.PureComponent {

  static propTypes = {
    client: PropTypes.object
  };

  state = {
    client: this.props.client
  };

  edit = (EO) => {
    voteEvents.emit('EditBtnClicked',this.props.client.id);
  }

  delete = (EO) => {
    voteEvents.emit('DeleteBtnClicked',this.props.client.id);
  }

  save = (EO) => {
    voteEvents.emit('SaveBtnClicked',this.props.client.id);
  }

  cancel = (EO) => {
    voteEvents.emit('CancelBtnClicked',this.props.client.id);
  }

  setSurnameRef = (ref) => {
    voteEvents.emit('setSurnameRef', ref, this.props.client.id);
  }

  setBallanceRef = (ref) => {
    voteEvents.emit('setBallanceRef', ref, this.props.client.id);
  }

  setNameRef = (ref) => {
    voteEvents.emit('setNameRef', ref, this.props.client.id);
  }

  setMiddleNameRef = (ref) => {
    voteEvents.emit('setMiddleNameRef', ref, this.props.client.id);
  }

  render() {
    var clientsCode = null;
    if(this.props.client){
      console.log("MobileClient id="+this.props.client.id+" render");
      clientsCode = 
        <tr className={this.props.client.id}>
          <td>
            {(this.props.client.disabledFlag) &&
              <input type='text' className='text' defaultValue={this.props.client.fam} ref={this.setSurnameRef} disabled/>
            }
            {(!this.props.client.disabledFlag ) &&
              <input type='text' className='text' defaultValue={this.props.client.fam} ref={this.setSurnameRef}/>
            }
          </td>
          <td>
            {(!this.props.client.addFlag) &&
                <input type='text' className='text' defaultValue={this.props.client.im} ref={this.setNameRef} disabled/>
            }
            {(this.props.client.addFlag) &&
                <input type='text' className='text' defaultValue={this.props.client.im} ref={this.setNameRef}/>
            }
          </td>
          <td>
            {(!this.props.client.addFlag) &&
              <input type='text' className='text' defaultValue={this.props.client.otch} ref={this.setMiddleNameRef} disabled/>
            }
            {(this.props.client.addFlag) &&
              <input type='text' className='text' defaultValue={this.props.client.otch} ref={this.setMiddleNameRef}/>
            }
          </td>
          <td>
            {(this.props.client.disabledFlag) &&
                <input type='text' className='text' defaultValue={this.props.client.balance} ref={this.setBallanceRef} disabled/>
            }
            {(!this.props.client.disabledFlag) &&
                <input type='text' className='text' defaultValue={this.props.client.balance} ref={this.setBallanceRef}/>
            }
          </td>
          {(this.props.client.state == 'active') && 
            <td className='StateActive'>
              <span className='State'>{this.state.client.state}</span>
            </td>
          }
          {(this.props.client.state == 'blocked') && 
            <td className='StateBlocked'>
              <span className='State'>{this.state.client.state}</span>
            </td>
          }   
          <td>
          {(this.props.client.disabledFlag) &&
              <input type='button' data-kind='edit' className='btnInGrid' value='Редактировать' onClick={this.edit}></input>
             //<MobileClientBtn dataKind='edit' className='btnInGrid' value='Редактировать' onClick={this.edit} />
          }
          {(!this.props.client.disabledFlag) &&
              <input type='button' className='btnInGrid' value='Сохранить' onClick={this.save}></input>
          }
          </td>
          <td>
          {(this.props.client.disabledFlag) &&
              <input type='button' kind='delete' className='btnInGrid' value='Удалить' onClick={this.delete}></input>
          }
          {(!this.props.client.disabledFlag) &&
              <input type='button' className='btnInGrid' value='Отменить' onClick={this.cancel}></input>
          }
          </td>
        </tr>;
    }
    return (clientsCode);
  }
}

export default MobileClient;
