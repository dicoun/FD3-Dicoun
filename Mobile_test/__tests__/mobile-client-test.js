import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from '../components/MobileClient';
import MobileClientBtn from '../components/MobileClientBtn.js';

test('работа MobileClient', () => {

  const client =  {id:101, fam:"Иванов", im:"Иван", otch:"Иванович", balance:'200', state: "active", disabledFlag: true};

  const component = renderer.create(
    <MobileClient key={client.id} client={client}/>
  );

  let componentTree=component.toJSON();

  // найдём в вёрстке компонента саму кнопку
  const editBtnElem = component.root.find( el => el.type=='input' && el.props.kind == 'edit' ); 

  // и "нажмём" на неё
  editBtnElem.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента саму кнопку
  const deleteBtnElem = component.root.find( el => el.type=='input' && el.props.kind == 'delete' ); 
  // и "нажмём" на неё
  deleteBtnElem.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
    
});
