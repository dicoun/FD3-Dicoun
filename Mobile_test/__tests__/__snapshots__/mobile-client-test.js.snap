// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileClient 1`] = `
<tr
  className={101}
>
  <td>
    <input
      className="text"
      defaultValue="Иванов"
      disabled={true}
      type="text"
    />
  </td>
  <td>
    <input
      className="text"
      defaultValue="Иван"
      disabled={true}
      type="text"
    />
  </td>
  <td>
    <input
      className="text"
      defaultValue="Иванович"
      disabled={true}
      type="text"
    />
  </td>
  <td>
    <input
      className="text"
      defaultValue="200"
      disabled={true}
      type="text"
    />
  </td>
  <td
    className="StateActive"
  >
    <span
      className="State"
    >
      active
    </span>
  </td>
  <td>
    <input
      className="btnInGridEdit"
      kind="edit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      className="btnInGridDel"
      kind="delete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа MobileClient 2`] = `
<tr
  className={101}
>
  <td>
    <input
      className="text"
      defaultValue="Иванов"
      disabled={true}
      type="text"
    />
  </td>
  <td>
    <input
      className="text"
      defaultValue="Иван"
      disabled={true}
      type="text"
    />
  </td>
  <td>
    <input
      className="text"
      defaultValue="Иванович"
      disabled={true}
      type="text"
    />
  </td>
  <td>
    <input
      className="text"
      defaultValue="200"
      disabled={true}
      type="text"
    />
  </td>
  <td
    className="StateActive"
  >
    <span
      className="State"
    >
      active
    </span>
  </td>
  <td>
    <input
      className="btnInGridEdit"
      kind="edit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      className="btnInGridDel"
      kind="delete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;
